client/state/posts/utils.js

```diff
@@ -144,7 +150,12 @@ export function normalizePostForDisplay( post ) {
 		return null;
 	}
 
-	return normalizeDisplayFlow( cloneDeep( post ) );
+	let normalizedPost = normalizePostCache.get( post );
+	if ( ! normalizedPost ) {
+		normalizedPost = normalizeDisplayFlow( cloneDeep( post ) );
```

Comment:
Curious why the use of `cloneDeep` here?

Reply:
`normalizeDisplayFlow` will modify the properties of the `post` passed as argument. For example, the `decodeEntities` step does things like:
```js
post.author.name = decode( post.author.name );
```
That's why we still need a deep copy.

Reply:
Hmm, that seems like probably something to avoid, but outside the scope of this PR.

Reply:
Maybe worth adding a comment that `normalizeDisplayFlow` will mutate its input, though?

Reply:
Comment added 😃